tm2/pkg/sdk/baseapp.go

```diff
@@ -409,12 +410,20 @@ func handleQueryApp(app *BaseApp, path []string, req abci.RequestQuery) (res abc
 			} else {
 				result = app.Simulate(txBytes, tx)
 			}
+
 			res.Height = req.Height
-			res.Value = amino.MustMarshal(result)
+
+			bytes, err := json.Marshal(result)
+			if err != nil {
+				res.Error = ABCIError(std.ErrInternal(fmt.Sprintf("cannot encode to JSON: %s", err)))
+			} else {
+				res.ResponseBase.Data = bytes
+			}
+
 			return res
 		case "version":
 			res.Height = req.Height
-			res.Value = []byte(app.appVersion)
+			res.ResponseBase.Data = []byte(app.appVersion)
```

Comment:
@piux2

Can you sanity check this?
I don't see an issue with having the version value be initialized in the response `Data`, rather than the `Value`

Reply:
Btw @jinoosss, why is this not marshalled into Amino?

Reply:
I added handling for changing the version response.
I'd appreciate it if you could check it out.
https://github.com/gnolang/gno/pull/3207/commits/d318f60f1b6e50047fa95112aa1f33cb28f14a7b